src/reanimated2/component/KeyboardAvoidingView.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { forwardRef, useCallback } from 'react';
+import type {
+  KeyboardAvoidingViewProps,
+  LayoutRectangle,
+  View,
+  ViewProps,
+} from 'react-native';
+import { StyleSheet, useWindowDimensions } from 'react-native';
+import { AnimatedView as ReanimatedView } from './View';
+import {
+  useAnimatedKeyboard,
+  useAnimatedStyle,
+  runOnUI,
+  useSharedValue,
+} from '../';
+
+let useHeaderHeight: () => number;
+try {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  useHeaderHeight = require('@react-navigation/elements').useHeaderHeight;
+} catch (e) {
+  useHeaderHeight = () => 0;
+}
+
+type Props = ViewProps & KeyboardAvoidingViewProps;
+
+/**
+ * View that moves out of the way when the keyboard appears by automatically
+ * adjusting its height, position, or bottom padding.
+ */
+const KeyboardAvoidingView = forwardRef<View, React.PropsWithChildren<Props>>(
+  (
+    {
+      behavior,
+      children,
+      contentContainerStyle,
+      enabled = true,
+      keyboardVerticalOffset = 0,
+      style,
+      onLayout,
+      ...props
+    },
+    ref
+  ) => {
+    const initialFrame = useSharedValue<LayoutRectangle | null>(null);
+    const keyboard = useAnimatedKeyboard();
+    const { height: screenHeight } = useWindowDimensions();
+    const headerHeight = useHeaderHeight();
+
+    const onLayoutWorklet = useCallback(
+      (layout: LayoutRectangle) => {
+        'worklet';
+        if (initialFrame.value == null) {
+          initialFrame.value = layout;
+        }
+      },
+      [initialFrame]
+    );
```

Comment:
I'm a bit lost. Why do we need `useCallback` here?

Reply:
So that `onLayoutWorklet` is a stable reference and can be used in a dependency array of `handleOnLayout`. In this case, `onLayoutWorklet` never changes because `initialFrame` is the only dependency and it's a shared value.

Reply:
I see, so [initialFrame] is passed as a dependency to satisfy the linter (and maybe extra safety).

Reply:
Kind of, it's indeed a dependency so we should regenerate the callback each time it changes but on the other hand it never changes.